src/store/modules/job/actions.ts

```diff
@@ -529,5 +539,51 @@ const actions: ActionTree<JobState, RootState> = {
     }
     return resp;
   },
+  async updateCurrentJob({ commit, state, dispatch }, payload) {
+    const cachedJobs = state.cached;
+    const pendingJobs = state.pending.list;
+
+    if(payload?.job) {
+      commit(types.JOB_CURRENT_UPDATED, payload.job);
+      return payload?.job;
+    }
+
+    const currentJob = pendingJobs.find((job: any) => job.jobId === payload.jobId) ? pendingJobs.find((job: any) => job.jobId === payload.jobId) : cachedJobs[payload?.jobId];
```

Comment:
We make currentJob let, assign it pendingJobs.find((job: any) => job.jobId === payload.jobId) and further it doesn't has value assign it cachedJobs[payload?.jobId]

Reply:
Sir, updated code as per comments.